### PR TITLE
fix useWatch defaultValue

### PR DESCRIPTION
**Pull request by bluebill1049**
FYI @keiya01 i think this will resolve issue 1 of that ticket.

here is the bug: https://codesandbox.io/s/react-hook-form-watch-default-values-ki6s8?file=/src/Watch.js

https://github.com/react-hook-form/react-hook-form/issues/1888

### CODE REVIEW

File: src/useWatch.ts [hunk 1]
```diff
@@ -35,12 +37,18 @@ export function useWatch<TWatchFieldValues>({
...
+    watchInternal,
+    defaultValuesRef,
+  } = control || methods.control;
   const [value, setValue] = React.useState<unknown>(
     isUndefined(defaultValue)
       ? isString(name)
-        ? defaultValue
+        ? get(defaultValuesRef.current, name)
+        : isArray(name)
+        ? name.map((inputName) => get(defaultValuesRef.current, inputName))
```
Comment (keiya01) [inline src/useWatch.ts:51]: I think `name.map` is not working correctly.

My suggestion is the following.

```ts
name.reduce((obj, inputName) => ({
  ...obj, 
  [inputName]: get(defaultValuesRef.current, inputName)
}), {})
```

**Review by keiya01 — COMMENTED** [review]
Thank you for fixing issue. But why is this PR Draft ?
This PR resolve issue 1.

File: src/useWatch.ts (hunk 1 above)
Comment (bluebill1049) [reply src/useWatch.ts:51]: aha, you are right! still, in the draft, because I still need to update those unit tests.

Comment (bluebill1049) [reply src/useWatch.ts:51]: i think we can save some bytes, later on, there are some common logic here with `watch`

Comment (keiya01) [reply src/useWatch.ts:51]: I understood.

Comment (keiya01) [reply src/useWatch.ts:51]: Yes, I think `useWatch` can  reduce code as well. There are some logic that is used in `watch`.

Comment (bluebill1049) [reply src/useWatch.ts:51]: 100% agree :)

**Comment by bluebill1049** [comment]
Screen Shot 2020-06-26 at 1 36 21 pm
@keiya01 i run test locally and get above, any idea.
`node v12.16.1`

**Comment by keiya01** [comment]
Are these bugs resolved on test ? I could not confirm these bugs...

**Comment by bluebill1049** [comment]
> Are these bugs resolved on test ? I could not confirm these bugs in local environment...

really? if you supply `defaultValues` at `useForm`, is `useWatch` return default value correctly?

**Comment by bluebill1049** [comment]
here is the bug: https://codesandbox.io/s/react-hook-form-watch-default-values-ki6s8?file=/src/Watch.js

**Comment by keiya01** [comment]
Yes... In my local env, all test passing.

**Comment by bluebill1049** [comment]
> Yes... In my local env, all test passing.

oh you refer to the `tests`, yeah strange... don't worry about it.

**Comment by keiya01** [comment]
Maybe node version is cause. My node version is v13.9.0 ...

**Comment by bluebill1049** [comment]
13 works 🥳 @keiya01 let me know if you are happy with this PR, we can merge it and save the code afterwards.

**Comment by keiya01** [comment]
@bluebill1049 LGTM🎉

**Comment by moulinraphael** [comment]
Thanks a lot @bluebill1049  !

**Pull request merged**